python/ray/_private/usage/usage_lib.py

```diff
@@ -277,6 +279,40 @@ def _put_extra_usage_tag(key: str, value: str):
         logger.debug(f"Failed to put extra usage tag, {e}")
 
 
+class _ThirdPartyLibraryUsageFinder(MetaPathFinder):
+    """Custom MetaPathFinder that logs the whitelisted 3rd party library usage."""
+
+    def find_spec(self, fullname, _, target=None):
+        library_name = self._get_library_name(fullname)
```

Comment:
can you try / catch the whole block so although there's a failure, it won't affect users?